### PR TITLE
Add Google Vertex AI Integration page

### PR DESCRIPTION
**Pull request by bilgeyucel**
Fixes https://github.com/deepset-ai/haystack-core-integrations/issues/120

### CODE REVIEW

File: integrations/google-vertex-ai.md
Comment (annthurium) [inline integrations/google-vertex-ai.md:65]: love this question

**Review by annthurium — APPROVED** [review]
test plan:
- [ ] made a copy of the Colab, ran through all the code examples to verify they produce the expected output

Nice work! Thorough and clear explanations and code. Everything looks good to me. I'm not sure if you also want a review from someone more tenured before merging.

File: integrations/google-vertex-ai.md
Comment (TuanaCelik) [inline integrations/google-vertex-ai.md:34]: It would be great to have a list of the components that this integration introduces as a rule in these integration pages. And then we can link the documentation page when they're ready. Example like the Gradient usage section: https://haystack.deepset.ai/integrations/gradient (adding links to docs now)

That's my only comment 🎉

**Comment by bilgeyucel** [comment]
I addressed the comments, merging this one

**Pull request merged**
